### PR TITLE
ftplugin(gdb): add Matchit support

### PR DESCRIPTION
**Pull request by Freed-Wu**

### CODE REVIEW

**Comment by chrisbra** [comment]
thanks. Can you please add a line like this one please: https://github.com/vim/vim/blob/9a91d2b72c20f213bbf77f27b7edd01e0e43d5e0/runtime/syntax/go.vim#L9

**Comment by Freed-Wu** [comment]
Fixed.

File: runtime/ftplugin/gdb.vim [hunk 1]
```diff
@@ -2,11 +2,19 @@
 " Language:	gdb
 " Maintainer:	Michaël Peeters <NOSPAMm.vim@noekeon.org>
 " Last Changed: 26 Oct 2017
+"               2024-03-17:          - add Matchit support (by Vim Project)
```
Comment (zeertzjq) [inline runtime/ftplugin/gdb.vim]:
```suggestion
"               2024-04-10:          - add Matchit support (by Vim Project)
```

Comment (Freed-Wu) [reply runtime/ftplugin/gdb.vim]: Fixed.

File: runtime/ftplugin/gdb.vim [hunk 2]
```diff
@@ -2,11 +2,19 @@
 " Language:	gdb
 " Maintainer:	Michaël Peeters <NOSPAMm.vim@noekeon.org>
 " Last Changed: 26 Oct 2017
+"               2024-04-10:          - add Matchit support (by Vim Project)
 
 if exists("b:did_ftplugin") | finish | endif
 let b:did_ftplugin = 1
 
 setlocal commentstring=#%s
+setlocal include=^\s*source
```
Comment (dkearns) [inline runtime/ftplugin/gdb.vim]:
```suggestion
setlocal include=^\\s*source
```

File: runtime/ftplugin/gdb.vim [hunk 3]
```diff
@@ -2,11 +2,19 @@
...
 
 " Undo the stuff we changed.
-let b:undo_ftplugin = "setlocal cms<"
+let b:undo_ftplugin = "setlocal cms< include<"
+
+" Matchit support
+if !exists('b:match_words')
+  let b:match_words = '\<if\|while\|define\|document\>:\<else\>:\<end\>'
```
Comment (dkearns) [inline runtime/ftplugin/gdb.vim]:
```suggestion
  let b:match_words = '\<\%(if\|while\|define\|document\)\>:\<else\>:\<end\>'
```

**Review by dkearns — CHANGES_REQUESTED** [review]

**Pull request closed without merging**